### PR TITLE
Morphological skeleton

### PR DESCRIPTION
**Pull request by 0x00b1**

### CODE REVIEW

File: cellprofiler/modules/morphologicalskeleton.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.morphology
+
+
+class Erosion(cellprofiler.module.Module):
+    category = "Mathematical morphology"
+    module_name = "Morphological skeleton"
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: `MorphologicalSkeleton`?

File: cellprofiler/modules/morphologicalskeleton.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.morphology
+
+
+class Erosion(cellprofiler.module.Module):
+    category = "Mathematical morphology"
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Do we want to simply call these modules `"Morphology"`?

File: cellprofiler/modules/morphologicalskeleton.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self.x_name = cellprofiler.setting.ImageNameSubscriber(
+            "Input"
+        )
+
+        self.y_name = cellprofiler.setting.ImageNameProvider(
+            "Output",
+            "OutputImage"
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Maybe `"Skeleton"`?

File: cellprofiler/modules/morphologicalskeleton.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        x_name = self.x_name.value
+        y_name = self.y_name.value
+
+        images = workspace.image_set
+
+        x = images.get_image(x_name)
+
+        x_data = x.pixel_data
+
+        y_data = skimage.morphology.skeletonize(x_data)
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: I think you need to use `skeletonize_3d`

File: cellprofiler/modules/morphologicalskeleton.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+        y = cellprofiler.image.Image(
+            image=y_data,
+            parent_image=x
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Set `dimensionality` of result.

File: cellprofiler/modules/morphologicalskeleton.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+        images.add(y_name, y)
+
+        if self.show_window:
+            workspace.display_data.x_data = x_data
+
+            workspace.display_data.y_data = y_data
+
+    def display(self, workspace, figure):
+        figure.set_grids((1, 2))
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Support 2D display.

File: cellprofiler/modules/morphologicalskeleton.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.morphology
+
+
+class Erosion(cellprofiler.module.Module):
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Rename `MorphologicalSkeleton`.

File: cellprofiler/modules/morphologicalskeleton.py (hunk 3 above)
Comment (0x00b1) [inline cellprofiler/modules/morphologicalskeleton.py]: @mcquin Go away

**Comment by mcquin** [comment]
Needs documentation and tests.

**Comment by mcquin** [comment]
Needs update

File: cellprofiler/modules/morphologicalskeleton.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
+"""
+
+Morphological skeleton thins an image into a single-pixel wide skeleton.
```
Comment (mcquin) [inline cellprofiler/modules/morphologicalskeleton.py]: Could you add a note that this is intended for use on binary images and label matrices?

**Pull request merged**